### PR TITLE
Rename snake_case functions in openrct2/src folder

### PR DESCRIPTION
**Pull request by Broxzier**

### CODE REVIEW

File: src/openrct2/Diagnostic.h [hunk 1]
```diff
@@ -70,20 +70,20 @@ enum class DiagnosticLevel
...
 #    define diagnostic_log_macro(level, format, ...)                                                                           \
-        diagnostic_log_with_location(level, __FILE__, __FUNCTION__, __LINE__, format, ##__VA_ARGS__)
+        DiagnosticLogWithLocation(level, __FILE__, __FUNCTION__, __LINE__, format, ##__VA_ARGS__)
 #else
 #    define diagnostic_log_macro(level, format, ...)                                                                           \
-        diagnostic_log_with_location(level, __FILE__, __func__, __LINE__, format, ##__VA_ARGS__)
+        DiagnosticLogWithLocation(level, __FILE__, __func__, __LINE__, format, ##__VA_ARGS__)
 #endif // _MSC_VER
 
 #define log_fatal(format, ...) diagnostic_log_macro(DiagnosticLevel::Fatal, format, ##__VA_ARGS__)
```
Comment (Gymnasiast) [inline src/openrct2/Diagnostic.h:85]: You missed these defines, which for all practical purposes are function names too.

Comment (Broxzier) [reply src/openrct2/Diagnostic.h:85]: I'll add those too.

File: src/openrct2-cli/Cli.cpp [hunk 2]
```diff
@@ -20,7 +20,7 @@ using namespace OpenRCT2;
 int main(int argc, const char** argv)
 {
     int32_t rc = EXIT_SUCCESS;
-    int runGame = cmdline_run(argv, argc);
+    int runGame = CmdLineRun(argv, argc);
```
Comment (Gymnasiast) [inline src/openrct2-cli/Cli.cpp:23]: My refactor, as well as the file name, use a lower case `l`.
I don’t mind one or the other, or indeed `CommandLine` (which is even better).

Comment (Broxzier) [reply src/openrct2-cli/Cli.cpp:23]: I agree with the full name, but let's leave that for later.

File: src/openrct2/Diagnostic.h [hunk 3]
```diff
@@ -70,20 +70,20 @@ enum class DiagnosticLevel
...
 extern bool _log_levels[static_cast<uint8_t>(DiagnosticLevel::Count)];
 
-void diagnostic_log(DiagnosticLevel diagnosticLevel, const char* format, ...);
-void diagnostic_log_with_location(
+void DiagnosticLog(DiagnosticLevel diagnosticLevel, const char* format, ...);
+void DiagnosticLogWithLocation(
     DiagnosticLevel diagnosticLevel, const char* file, const char* function, int32_t line, const char* format, ...);
 
 #ifdef _MSC_VER
 #    define diagnostic_log_macro(level, format, ...)                                                                           \
```
Comment (Gymnasiast) [inline src/openrct2/Diagnostic.h:78]: Let’s also do `diagnostic_log_macro`, for the same reasons.

File: src/openrct2-cli/Cli.cpp (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2-cli/Cli.cpp:23]: https://github.com/OpenRCT2/OpenRCT2/issues/19176

**Review by Gymnasiast — APPROVED** [review]

**Comment by Gymnasiast** [comment]
I rebased your PR, but it seems there are still some issues with Android.

**Comment by duncanspumpkin** [comment]
I've had to drop that last commit and rework it so that the macros are SHOUTING_SNAKE_CASE to avoid the name collisions.

**Comment by Broxzier** [comment]
Looks like a conflict with an actual `LogError` function. I'm not sure if we should give macros the same name, since they work differently from function calls. Usually, they are all caps for this reason.

**Comment by duncanspumpkin** [comment]
> Looks like a conflict with an actual `LogError` function. I'm not sure if we should give macros the same name, since they work differently from function calls. Usually, they are all caps for this reason.

yes i've gone for LOG_ERROR for the macros as that matches the general C++ coding style for macros.

**Comment by Gymnasiast** [comment]
I'm not entirely happy about that, though, as we use them as functions. I'd much prefer using the same casing scheme.

**Comment by Broxzier** [comment]
It's true they look like functions, but they are not. That's only all the more reason to go for a different naming convention for them. All capt with snake case is the standard choice here.

I agree that they should be functions eventually, but until we make a switch to C++20 (where we can use `<source_location>`, which is not yet fully supported on all compilers), there is unfortunately no way to turn them into functions without losing the source file and line information.

**Comment by duncanspumpkin** [comment]
> I'm not entirely happy about that, though, as we use them as if they were functions (and they might be redefined as constexpr or inline functions some day). I'd much prefer using the same casing scheme.

I'm much the same but it's just not going to work atm without making them different in some way. Can't even do :: to access it.

**Comment by Gymnasiast** [comment]
> It's true they look like functions, but they are not. That's only all the more reason to go for a different naming convention for them. All caps with snake case is the standard choice here.
> 
> I agree that they should be functions eventually, but until we make a switch to C++20 (where we can use `<source_location>`, which is not yet fully supported on all compilers), there is unfortunately no way to turn them into functions without losing the source file and line information.

I cannot say that I agree, but I'm not going to block the PR over it.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**